hazelcast/src/main/java/com/hazelcast/map/impl/MapServiceContextImpl.java

```diff
@@ -427,6 +429,8 @@ public void destroyMap(String mapName) {
         // Statistics are destroyed after container to prevent their leak.
         destroyPartitionsAndMapContainer(mapContainer);
```

Comment:
nit: Could you also move this `destroyPartitionsAndMapContainer` call to the bottom since it looks like we're using the destroyed object later? (I understand that we don't harm its content with destroy, just removing its references from the related places and I am aware that it will not be collected since we already have a reference to it in this method but it seems ugly to me.)

Reply:
I think, current form is safer, since we are removing an already removed map-container's stuff, otherwise memory leak possibility is higher if we receive event listener before map-container removal.